pipeline/load.py

```diff
@@ -0,0 +1,288 @@
+            logging.info(
+                "'%s' already exists with release ID %s.", release_name, release_id[0])
+            return release_id[0]
+
+        insert_query = """INSERT INTO release (release_name, release_date, artist_id, type_id)
+                            VALUES (%s, %s, %s, %s)
+                            RETURNING release_id;"""
+        cursor.execute(insert_query, (release_name,
+                                      release_date, artist_id, type_id))
+        release_id = cursor.fetchone()
+        logging.info(
+            "Release '%s' of '%s' was added to the database.", release_id, release_name)
+        return release_id[0]
+
+    except psycopg2.Error:
+        logging.error("Error inserting release: %s", release_name)
+        return -1
+
+
+def insert_release_genres(release_id: int, genre_id: int, genre_name: str,
+                          release_name: str, cursor: psycopg2.extensions.cursor) -> None:
+    """
+    Inserts release genres into the database if it doesn't already exist.
+    Logs whether the release genre was added or already exists.
+    """
+    try:
+        check_query = """SELECT EXISTS
+                        (SELECT 1 FROM release_genre
+                        WHERE release_id = %s
+                        AND genre_id = %s);"""
+
+        cursor.execute(check_query, (release_id, genre_id))
+        exists = cursor.fetchone()[0]
+
+        if exists:
+            logging.info(
+                "Release '%s' already has '%s' as a genre.", release_name, genre_name)
+        else:
+            insert_query = """INSERT INTO release_genre (release_id, genre_id)
+                                VALUES (%s, %s);"""
+            cursor.execute(insert_query, (release_id, genre_id))
+            logging.info(
+                "'%s' was added to '%s' successfully.", genre_name, release_name)
+    except psycopg2.Error:
+        logging.error(
+            "Error adding '%s' to '%s' ID: %s", genre_name, release_name, release_id)
+
+
+def main_load(sales_df: pd.DataFrame) -> None:
+    """
+    Loads data from the sales dataframe into the database.
+    """
+    config_log()
+    connection = get_connection()
+    cursor = get_cursor(connection)
+
+    sales_df["genres"] = sales_df["genres"].apply(ast.literal_eval)
+
+    for _, row in sales_df.iterrows():
+        insert_country(row["country"], cursor)
+        insert_artist(row["artist_name"], row["country"], cursor)
+
+        release_id = insert_release(row["release_name"], row["release_date"],
+                                    row["release_type"], row["artist_name"], cursor)
+
+        for genre in row["genres"]:
+            genre_id = insert_genres(genre.lower(), cursor)
+            if genre_id == -1:
+                continue
+            insert_release_genres(release_id, genre_id,
+                                  genre.lower(), row["release_name"], cursor)
+
+    connection.commit()
+
+
+if __name__ == "__main__":
+    pass
```

Comment:
probably want main_load() here

Reply:
Yeah we want the load script to be able to be ran independently as well as via the main pipeline

Reply:
The issue with having the function there is the lack of argument. If it's a standalone script that we're not using a .csv for, there's nothing for me to put in.